fhir-path/src/main/java/com/ibm/fhir/path/evaluator/FHIRPathEvaluator.java

```diff
@@ -1457,5 +1462,59 @@ public void unsetExternalConstant(String name) {
         public boolean hasExternalConstant(String name) {
             return externalConstantMap.containsKey(name);
         }
+
+        /**
+         * Set the constraint currently under evaluation
+         *
+         * @param constraint
+         *     the constraint currently under evaluation
+         */
+        public void setConstraint(Constraint constraint) {
```

Comment:
when is setting the constraint required?  when is it not required?

Reply:
It is driven by the Validator to give the FHIRPathEvaluator and any FHIRPathFunction(s) access to it during evaluation. Thus far, the only function that cares about it is MemberOfFunction.

Reply:
new method doc looks good